compiler/modules/front/expr.bal

```diff
@@ -574,6 +580,29 @@ function codeGenArithmeticBinaryExpr(ExprContext cx, bir:BasicBlock bb, bir:Arit
     return { result, block: bb };
 }
 
+function isNonPanickingIntOp(bir:Operand lhs, bir:Operand rhs, bir:ArithmeticBinaryOp op) returns boolean {
+    t:IntSubtypeConstraints? lhsConstraints = t:intSubtypeConstraints(lhs.semType);
+    t:IntSubtypeConstraints? rhsConstraints = t:intSubtypeConstraints(rhs.semType);
+    if lhsConstraints == () || rhsConstraints == () {
+        return false;
+    }
+    match op {
+        "+"|"-" => {
+            // largest type that can't overflow is unsigned32 (positive) and signed32 (negative)
+            return lhsConstraints.max <= 0xffffffff && rhsConstraints.max <= 0xffffffff &&
+                   lhsConstraints.min >= -0x80000000 && rhsConstraints.min >= -0x80000000;
+        }
+        "*" => {
+            // largest type that can't overflow is unsigned16 (positive) and signed16 (negative)
+            return lhsConstraints.max <= 0xffff && rhsConstraints.max <= 0xffff &&
```

Comment:
Really? Can two int:Signed32 values overflow? Which?

Reply:
Sorry I didn't notice that there is a typo in the [spec](https://ballerina.io/spec/lang/2021R1/#built-in_subtypes) (positive side have one digit more than the negative side).

Reply:
created issue #1038